apps/chatter/consumers.py

```diff
@@ -88,31 +90,43 @@ async def error_reponse(self, error):
             'command': self.ERROR,
             'error': error
         }
-        self.send_message(content)
+        await self.send_message(content)
 
     commands = {
         INIT_CHAT: init_chat,
-        FETCH_MESSAGES: fetch_messages,
-        NEW_MESSAGE: new_message
+        FETCH_ENTRIES: fetch_entries,
+        NEW_ENTRY: new_entry
     }
 
     async def receive(self, text_data):
-        print('IN RECEIVE WITH TEXT DATA ' + text_data)
-        data = json.loads(text_data)
-        if data['command'] == ChatConsumer.INIT_CHAT:
-            await self.init_chat(data)
-        elif data['command'] == ChatConsumer.FETCH_MESSAGES:
-            await self.fetch_messages(data)
-        elif data['command'] == ChatConsumer.NEW_MESSAGE:
-            await self.new_message(data)
+        print('IN RECEIVE WITH MESSAGE STRING ' + text_data)
+        message = json.loads(text_data)
+        message_serializer = MessageSerializer(data=message)
+
+        if message_serializer.is_valid():
+            print('VALID SERIALIZER: ', message_serializer.validated_data)
+            validMessage = message_serializer.validated_data
+            if validMessage['command'] == ChatConsumer.INIT_CHAT:
+                await self.init_chat(validMessage)
+            elif validMessage['command'] == ChatConsumer.FETCH_ENTRIES:
+                await self.fetch_entries(validMessage)
+            elif validMessage['command'] == ChatConsumer.NEW_ENTRY:
+                await self.new_entry(validMessage)
+            else:
+                await self.error_reponse('No procedure is available for command \"{0}\"'
+                    .format(validMessage['command']))
         else:
-            await self.error_reponse('The command \"{0}\" could not be recognized'
-                .format(data['command']))
+            await self.error_reponse('The data sent could not be recognized')
+            print('INVALID SERIALIZER: ', message_serializer.errors)
+        
 
+    async def send_message(self, message):
```

Comment:
You have `async def send_message` twice.

Reply:
I shall vanquish this